### PR TITLE
Deactivate 'Previous Step' button at first level of derivation

### PR DESCRIPTION
**Pull request by flatFeather**
Hide "Previous Step" button when the current screen is the first derivation step and there is no previous step. Currently the button appears but when clicked it does not perform any action.

### CODE REVIEW

**Comment by MuhammadMuradG** [comment]
Good observation, but from UI perspective I thinks it will be great if we disable the button instead of hidden it.

File: src/greatwall/gui.py [hunk 1]
```diff
@@ -1002,6 +1006,9 @@ def config_selecting_derivation_widgets(self):
             ):
                 if idx == 0:
                     selection_button = widgets
+                    selection_button.setEnabled(
+                        False if self.greatwall.current_level == 0 else True
```
Comment (MuhammadMuradG) [inline src/greatwall/gui.py:1010]: I add code formating to match our current formating.  However, we intend to support one of the code formating tool, like black, to simplifying this process.

**Review by MuhammadMuradG — APPROVED** [review]
It seems good now, I add commits for code formating and to fix shape generation in derivation view.

**Review by Yuri-SVB — APPROVED** [review]

**Comment by Yuri-SVB** [comment]
Branch protection bypassed due to manual review.
FlatFeather is henceforward required to sign commits for strict adherence to protocol.

**Pull request merged**